### PR TITLE
Export metric about schema feature use

### PR DESCRIPTION
**Pull request by msullivan**
Currently report whether the following are used:
 - policies
 - triggers
 - rewrites
 - globals
 - computed globals
 - aliases
 - functions
 - computed pointers
 - FTS

### CODE REVIEW

File: edb/server/metrics.py [hunk 1]
```diff
@@ -200,6 +200,12 @@
     labels=('tenant', 'extension'),
 )
 
+feature_used = registry.new_labeled_gauge(
+    'feature_used_branch_count_current',
+    'How many branches a schema feature is used by.',
+    labels=('tenant', 'feature'),
+)
```
Comment (1st1) [inline edb/server/metrics.py:207]: @zackelan please confirm we can plot this later

File: edb/server/compiler/ddl.py [hunk 2]
```diff
@@ -1169,6 +1184,53 @@ def _reset_schema(
     )
 
 
+_FEATURE_NAMES: dict[type[s_obj.Object], str] = {
+    s_policies.AccessPolicy: 'policy',
+    s_triggers.Trigger: 'trigger',
+    s_rewrites.Rewrite: 'rewrite',
+    s_globals.Global: 'global',
+    s_expraliases.Alias: 'alias',
+    s_func.Function: 'function',
```
Comment (1st1) [inline edb/server/compiler/ddl.py:1193]: Not necessarily for this PR, but maybe we can add:

* link properties
* more than one base type on types
* annotations
* indexes (i'd like to see how often people define them!)
* constraints (ditto)
* computed properties (ditto)
* computed links
* multi properties
* enums

**Review by 1st1 — APPROVED** [review]
Amazing stuff, I can't wait until we start seeing these stats.

File: edb/server/metrics.py (hunk 1 above)
Comment (zackelan) [reply edb/server/metrics.py:207]: yes, this fits well into the Prometheus data model, so shouldn't pose any problems with aggregating it after deployment

File: edb/server/compiler/ddl.py (hunk 2 above)
Comment (msullivan) [reply edb/server/compiler/ddl.py:1193]: @zackelan I can't tell if I'm missing something here: is there are major difference between what you are suggesting we do and what I'm doing here?

Comment (msullivan) [reply edb/server/compiler/ddl.py:1201]: @1st1 I added all those (a third or so were already there)

Comment (zackelan) [reply edb/server/compiler/ddl.py:1201]: nothing in particular - just a +1 to Yury's suggestion of adding metrics like Oprah giving away cars

**Pull request merged**